client/lib/currency-formatter/test/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import currencyFormatter, { findCurrency } from 'lib/currency-formatter';
+
+describe( 'currencyFormatter', () => {
+	it( 'formats a number to localized currency', () => {
+		const money = currencyFormatter( 99.32, { code: 'USD' } );
+		expect( money ).to.equal( '$99.32' );
+	} );
+	it( 'adds a localized thousands separator', () => {
+		const money = currencyFormatter( 9800900.32, { code: 'USD' } );
+		expect( money ).to.equal( '$9,800,900.32' );
+	} );
+
+	describe( 'supported currencies', () => {
+		it( 'USD', () => {
+			const money = currencyFormatter( 9800900.32, { code: 'USD' } );
+			expect( money ).to.equal( '$9,800,900.32' );
+		} );
+		it( 'AUD', () => {
+			const money = currencyFormatter( 9800900.32, { code: 'AUD' } );
+			expect( money ).to.equal( 'A$9,800,900.32' );
+		} );
+		it( 'CAD', () => {
+			const money = currencyFormatter( 9800900.32, { code: 'CAD' } );
+			expect( money ).to.equal( 'C$9,800,900.32' );
+		} );
+		it( 'EUR', () => {
+			const money = currencyFormatter( 9800900.32, { code: 'EUR' } );
+			expect( money ).to.equal( '9 800 900,32 €' );
```

Comment:
This depends on the country, some will use spaces for the thousand separator. We can still update the default here. https://docs.oracle.com/cd/E19455-01/806-0169/overview-9/index.html